selfservice/strategy/saml/strategy_login.go

```diff
@@ -0,0 +1,156 @@
+package saml
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/ory/herodot"
+	"github.com/ory/kratos/continuity"
+	"github.com/ory/kratos/identity"
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/session"
+	"github.com/ory/kratos/text"
+	"github.com/ory/kratos/ui/node"
+	"github.com/ory/kratos/x"
+)
+
+// Implement the interface
+var _ login.Strategy = new(Strategy)
+
+// Call at the creation of Kratos, when Kratos implement all authentication routes
+func (s *Strategy) RegisterLoginRoutes(r *x.RouterPublic) {
+	s.setRoutes(r)
+}
+
+// SubmitSelfServiceLoginFlowWithSAMLMethodBody is used to decode the login form payload
+// when using the saml method.
+//
+// swagger:model SubmitSelfServiceLoginFlowWithSAMLMethodBody
+type SubmitSelfServiceLoginFlowWithSAMLMethodBody struct {
+	// The provider to register with
+	//
+	// required: true
+	Provider string `json:"samlProvider"`
+
+	// The CSRF Token
+	CSRFToken string `json:"csrf_token"`
+
+	// Method to use
+	//
+	// This field must be set to `oidc` when using the oidc method.
+	//
+	// required: true
+	Method string `json:"method"`
+
+	// The identity traits. This is a placeholder for the registration flow.
+	Traits json.RawMessage `json:"traits"`
+}
+
+// Login and give a session to the user
+func (s *Strategy) processLogin(w http.ResponseWriter, r *http.Request, a *login.Flow, provider Provider, c *identity.Credentials, i *identity.Identity, claims *Claims) (*registration.Flow, error) {
+
+	s.updateIdentityTraits(w, r, i, provider, claims)
```

Comment:
error ignored

Reply:
this is basically ineffective because we never save the traits updates. The data is also not being applied correctly because we're lacking validation.